src/javascript/_common/base/subscription_manager.js

```diff
@@ -63,6 +63,45 @@ const SubscriptionManager = (() => {
         }
     };
 
+    /**
+     * Add subscription without subscribers from request
+     * E.g. open subscription to proposal_open_contract on buy request
+     * @param {String}   msg_type               msg_type of the subscription
+     * @param {Object}   send_request           the object of the request to be made
+     * @param {Object}   subscribe_request      the object of the subscription request
+     * @param {String}   subscription_prop      prop to add to subscribe_request from initial request, e.g. contract_id
+     */
+    const addSubscriptionFromRequest = (msg_type, send_request, subscribe_request, subscription_prop = '') =>
+        new Promise((resolve, reject) => {
+            let sub_id;
+            let is_stream = false;
+
+            BinarySocket.send(send_request, {
```

Comment:
@OskarAhl Shouldn't it manage to prevent from sending duplicate buy requests and just subscribes them to the current subscription?

Reply:
@morteza-binary Should it? Currently it's the same buy as the buy in the `ws-methods.js`, and the purpose is to add a subscription without any subscribers - since the subscription is implicitly added as a side effect from sending the buy request.

Reply:
so it's fine.